tests/validators/tests.py

```diff
@@ -325,19 +324,209 @@
+    'http://example.',
+    'http://example.com:-1',
+    'http://example.com:-1/',
+    'http://example.com:000000080',
+    'http://example.com:000000080/',
+    'http://.com',
+    'http://invalid-.com',
+    'http://-invalid.com',
+    'http://invalid.com-',
+    'http://invalid.-com',
+    'http://inv-.alid-.com',
+    'http://inv-.-alid.com',
+    'file://localhost/path',
+    'git://example.com/',
+    'http://.',
+    'http://..',
+    'http://../',
+    'http://?',
+    'http://??',
+    'http://??/',
+    'http://#',
+    'http://##',
+    'http://##/',
+    'http://foo.bar?q=Spaces should be encoded',
+    '//',
+    '//a',
+    '///a',
+    '///',
+    'http:///a',
+    'foo.com',
+    'rdar://1234',
+    'h://test',
+    'http:// shouldfail.com',
+    ':// should fail',
+    'http://foo.bar/foo(bar)baz quux',
+    'http://-error-.invalid/',
+    'http://dashinpunytld.trailingdot.xn--.',
+    'http://dashinpunytld.xn---',
+    'http://-a.b.co',
+    'http://a.b-.co',
+    'http://a.-b.co',
+    'http://a.b-.c.co',
+    'http:/',
+    'http://',
+    'http://',
+    'http://1.1.1.1.1',
+    'http://123.123.123',
+    'http://3628126748',
+    'http://123',
+    'http://000.000.000.000',
+    'http://016.016.016.016',
+    'http://192.168.000.001',
+    'http://01.2.3.4',
+    'http://01.2.3.4',
+    'http://1.02.3.4',
+    'http://1.2.03.4',
+    'http://1.2.3.04',
+    'http://.www.foo.bar/',
+    'http://.www.foo.bar./',
+    'http://[::1:2::3]:8/',
+    'http://[::1:2::3]:8080/',
+    'http://[]',
+    'http://[]:8080',
+    'http://example..com/',
+    'http://aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa.example.com',
+    'http://example.aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa.com',
+    'http://example.aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa',
+    'http://aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa.aaaaaaaaaaaaaaaaaaaaaaaaaaaaa.aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa.aaaaaa\
+     aaaaaaaaaaaaaaaaaaaaaaaa.aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa.aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa.aaaaaaaaaaaaaaaaaaaaaaa\
+     aaaaaaa.aaaaaaaaaaaaaaaaaaaaaaaaa',
+    'https://test.[com',
+    'http://@example.com',
+    'http://:@example.com',
+    'http://:bar@example.com',
+    'http://foo@bar@example.com',
+    'http://foo/bar@example.com',
+    'http://foo:bar:baz@example.com',
+    'http://foo:bar@baz@example.com',
+    'http://foo:bar/baz@example.com',
+    'http://invalid-.com/?m=foo@example.com    ',
+]
 
 # Add valid and invalid URL tests.
 # This only tests the validator without extended schemes.
-with open(create_path('valid_urls.txt'), encoding='utf8') as f:
-    for url in f:
-        TEST_DATA.append((URLValidator(), url.strip(), None))
-with open(create_path('invalid_urls.txt'), encoding='utf8') as f:
-    for url in f:
-        TEST_DATA.append((URLValidator(), url.strip(), ValidationError))
+for url in VALID_URLS:
+    TEST_DATA.append((URLValidator(), url.strip(), None))
```

Comment:
Also, you can use `extend()` with a generator expression to eliminate the need for a `for` loop with indentation.

Reply:
Sure.